spec/ruby_version_specific/contracts_spec_2.1.rb

```diff
@@ -1,6 +1,6 @@
 class GenericExample
-  Contract String, Bool, Args[Symbol], Float, { e: Range, f: Maybe[Num] }, Proc =>
-           [Proc, Hash, Num, Range, Float, ArrayOf[Symbol], Bool, String]
+  Contract String, Bool, Args[Symbol], Float, OptHash[e: Range, f: Opt[Num]], Proc =>
```

Comment:
It is already tested by this suite - look at the full file, it is the first example.